axelor-production/src/main/java/com/axelor/apps/production/service/BillOfMaterialServiceImpl.java

```diff
@@ -449,6 +449,49 @@ public List<BillOfMaterial> getAlternativesBOM(Product originalProduct, Company
         .fetch();
   }
 
+  protected BillOfMaterial getAnyBOM(Product originalProduct, Company company) {
+    return billOfMaterialRepo
+        .all()
+        .filter(
+            "self.product = ?1 AND self.company = ?2 AND self.statusSelect = ?3",
+            originalProduct,
+            company,
+            BillOfMaterialRepository.STATUS_APPLICABLE)
+        .order("id")
+        .fetchOne();
+  }
+
+  @Override
+  public BillOfMaterial getBOM(Product originalProduct, Company company) throws AxelorException {
+
+    if (company == null) {
+      company = AuthUtils.getUser().getActiveCompany();
```

Comment:
be careful, AuthUtils.getUser() can return null, and the activeCompany on an user can also be null.